libs/components/behavioral/src/lib/footer-behavior/footer-behavior.component.ts

```diff
@@ -0,0 +1,61 @@
+import { CommonModule } from '@angular/common';
+import { ChangeDetectionStrategy, Component, inject } from '@angular/core';
+import { MatDialog, MatDialogModule } from '@angular/material/dialog';
+import { dispatch, selectQuerySnapshot } from '@hra-ui/cdk/injectors';
+import { ResourceRegistrySelectors as RR } from '@hra-ui/cdk/state';
+import { FooterComponent } from '@hra-ui/components/molecules';
+import { LinkIds, ResourceIds as Ids, DownloadActions, DownloadFormat } from '@hra-ui/state';
+
+import { ContactBehaviorComponent } from '../contact-behavior/contact-behavior.component';
+
+/** A component for footer behavior which provides functionality for various buttons in footer */
+@Component({
+  selector: 'ftu-footer-behavior',
+  standalone: true,
+  imports: [CommonModule, MatDialogModule, FooterComponent, ContactBehaviorComponent],
+  templateUrl: './footer-behavior.component.html',
+  styleUrls: ['./footer-behavior.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class FooterBehaviorComponent {
+  /** Input for product logo URL to displayed on the left side. */
+  readonly productLogoUrl = selectQuerySnapshot(RR.url, Ids.ProductLogoUrl);
+
+  /** Input for product title to displayed on the left side. */
+  readonly productTitle = selectQuerySnapshot(RR.anyText, Ids.ProductTitle);
+
+  /** Hra portal link routing after clicking on HRA Portal */
+  readonly hraPortal = LinkIds.Portal;
+
+  /** Metadata page link routing after clicking on Illustration */
+  readonly illustrationMetadata = LinkIds.Illustration;
+
+  /** Embed link routing after clicking on Embed */
+  readonly embed = LinkIds.Embed;
+
+  /** A dialog box which shows contact modal after clicking on contact */
+  private readonly dialog = inject(MatDialog);
+
+  /** A dispatcher function to download file in specified format */
+  readonly download = dispatch(DownloadActions.Download);
+
+  // /** Different download formats options displayed to the user */
+  // readonly downloadFormats = [
+  //   { label: 'Download XLXS' },
+  //   { label: 'Download JSON-LD' },
+  //   { label: 'Download PDF' },
+  //   { label: 'Download AI' },
+  //   { label: 'Download PNG' },
+  // ];
+
+  /** A function which opens the contact modal dialog box */
+  contact(): void {
+    this.dialog.open(ContactBehaviorComponent);
+  }
+
+  /** A function which opens download list popup */
+  downLoadClicked(format: DownloadFormat) {
+    // console.log('Method not implemented.' + event);
+    this.download(format);
```

Comment:
Call `download` directly from the html instead of wrapping it in a pointless method